packages/office-ui-fabric-react/src/components/CommandBar/examples/CommandBar.ButtonAs.Example.tsx

```diff
@@ -22,19 +25,20 @@ const menuStyles: Partial<IContextualMenuStyles> = {
   subComponentStyles: { menuItem: itemStyles, callout: {} },
 };
 
+const getCommandBarButtonStyles = memoizeFunction(
```

Comment:
Can you use `React.useMemo` here?

Reply:
`React.useMemo` will reset the cache when component unmount. so here - if `React.useMemo` is used and `CustomButton ` gets mounted 100 times with the same styles object as prop, each `CustomButton` render will get a new `styles` objects which will lead to 100 classname calculations.

Reply:
Oh interesting. Can you add a comment about that in the example?

Reply:
it's on my TODO to write a short wiki about good/bad styles prop in general. i think this is definitely worth noting

Reply:
I think it might still be worth adding a comment specifically in this instance, when in general using useMemo would seem reasonable.

Reply:
https://github.com/microsoft/fluentui/wiki/Performance